### DISCUSSION TITLE
window.open( ) is not working when browser is inactive state

### DISCUSSION THREAD

**Discussion by ZoraizIBS** [community/community#120712] [Programming Help]
### Select Topic Area

Product Feedback

### Body

basically, my problem is that i have an application the app can b read RSS feeds and display them on my page or screen. Also i have used the timer to reload the component of rss feed. I have set some matching keywords like if some keywords match with rss feed result then it will open in a new tab. This works fine when the browser is active but when i am working on any other task or open another window and my browser goes to an inactive state . So now i want to fix this issue how i can fix this ? When the browser is in active the app is not opens the new tabs.

My Code Is :
```

foreach (var item in Model.Items.OrderByDescending(a => a.PublishDate))
    {
       
                lebel = "New";
                bool ismatch = CommonFunctions.KeyWordsMatches(item, KeyWordsArray, LebelsArray, _settings.Value);
                if (ismatch)
                {
                    syndicationItemList.Add(item);
                     <script type="text/javascript">
                    window.open("@item.Links.FirstOrDefault().Uri", '_blank');
                </script>
                }
            
        

        <button type="button" class="btn btn-primary" data-toggle="collapse" data-target="#demo_@rowCount">
            @rowCount -  @item.Title.Text.ToString()  <b style="color:red"> @lebel</b>  - <b style="color:aliceblue">  @timeAgo</b>
        </button>
        <div id="demo_@rowCount" class="collapse">
            <p>Link: <a href="@item.Links.FirstOrDefault().Uri.AbsoluteUri" target="_blank">@item.Links.FirstOrDefault().Uri.AbsoluteUri</a></p>
            <p>Publish Date: @item.PublishDate.ToString()</p>
            <p>Description: @Html.Raw(item.Summary.Text)</p>
            
        </div>
       
       
    }
```

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @ZoraizIBS !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Feed category is solely related to conversations around the GitHub product Feed. This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!

**Comment by ebndev** [comment]
Hey @ZoraizIBS!

Front-end applications tend to only run scripts when the page is active. To achieve what you're looking for, you'd most likely have to create a back-end application- probably using Node since it looks like you're familiar with Javascript.

You can run the app on your local machine or deploy + host it online, then app can then continuously listen for RSS feed updates and take action / send notifications.

Here's an example of a project: [How to send browser push notifications from a Node.js app](https://knock.app/blog/how-to-send-browser-push-notifications-from-nodejs). I'm thinking you could adjust the code to notify you when a new result is found and also open the result link in a new tab.

Best of luck!
